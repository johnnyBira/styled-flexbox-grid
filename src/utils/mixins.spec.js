import {
  getVerticalAlignment,
  getHorizontalAlignment,
  verticalAlignSelf,
  order,
  span,
  offset,
  gutter,
  justify,
  direction,
  alignColumns,
  verticalAlignColumns,
  sideMargin,
} from './mixins';

describe('Grid/mixins', () => {
  describe('getVerticalAlignment', () => {
    it('returns the corrent flex alignemnt attribute', () => {
      expect(getVerticalAlignment('center')).toEqual('center');
      expect(getVerticalAlignment('bottom')).toEqual('flex-end');
      expect(getVerticalAlignment('top')).toEqual('flex-start');
    });
    it('returns `normal` as the default flex alignemnt attribute', () => {
      expect(getVerticalAlignment('test')).toEqual('normal');
    });
  });

  describe('getHorizontalAlignment', () => {
    it('returns the corrent flex alignemnt attribute', () => {
      expect(getHorizontalAlignment('center')).toEqual('center');
      expect(getHorizontalAlignment('right')).toEqual('flex-end');
      expect(getHorizontalAlignment('left')).toEqual('flex-start');
    });
    it('returns `normal` as the default flex alignemnt attribute', () => {
      expect(getHorizontalAlignment('test')).toEqual('normal');
      expect(getHorizontalAlignment()).toEqual('normal');
    });
  });

  describe('verticalAlignSelf', () => {
    it('returns the css attribute `align-self` with the provided value', () => {
      expect(verticalAlignSelf('center')).toMatchSnapshot();
    });
  });

  describe('order', () => {
    it('returns the css attribute `order` with the provided value', () => {
      expect(order(1)).toMatchSnapshot();
    });
  });

  describe('columns', () => {
    it('returns the css attribute `max-width` with the provided value', () => {
      expect(span('50%')).toMatchSnapshot();
    });
  });

  describe('offset', () => {
    it('returns the css attribute `margin-left` with the provided value', () => {
      expect(offset('50%')).toMatchSnapshot();
    });
  });

  describe('gutter', () => {
    it('returns the css attributes `padding-left` and `padding-right` with the provided value', () => {
      expect(gutter('1rem')).toMatchSnapshot();
    });
  });

  describe('justify', () => {
    it('returns the css attribute `justify-content` with the provided value', () => {
      expect(justify('end')).toMatchSnapshot();
      expect(justify('start')).toMatchSnapshot();
      expect(justify('center')).toMatchSnapshot();
      expect(justify('around')).toMatchSnapshot();
      expect(justify('between')).toMatchSnapshot();
      expect(justify('evenly')).toMatchSnapshot();
    });
    it('returns `normal` as the default `justify-content` attribute', () => {
      expect(justify('test')).toMatchSnapshot();
      expect(justify()).toMatchSnapshot();
    });
  });

  describe('direction', () => {
    it('returns the css attribute `flex-direction` with the provided value', () => {
      expect(direction('column')).toMatchSnapshot();
      expect(direction('columnReverse')).toMatchSnapshot();
      expect(direction('row')).toMatchSnapshot();
      expect(direction('rowReverse')).toMatchSnapshot();
      expect(direction('reverse')).toMatchSnapshot();
      expect(direction('unset')).toMatchSnapshot();
      expect(direction('initial')).toMatchSnapshot();
    });
    it('returns `normal` as the default `flex-direction` attribute', () => {
      expect(direction('test')).toMatchSnapshot();
      expect(direction()).toMatchSnapshot();
    });
  });

  describe('alignColumns', () => {
    it('returns the css attribute `justify-content` with the provided value', () => {
      expect(alignColumns('center')).toMatchSnapshot();
    });
  });

  describe('verticalAlignColumns', () => {
    it('returns the css attribute `align-items` with the provided value', () => {
      expect(verticalAlignColumns('center')).toMatchSnapshot();
    });
  })
});
