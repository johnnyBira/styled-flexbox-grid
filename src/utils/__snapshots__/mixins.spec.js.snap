// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid/mixins alignColumns returns the css attribute \`justify-content\` with the provided value 1`] = `
"
    justify-content: center;
  "
`;

exports[`Grid/mixins columns returns the css attribute \`max-width\` with the provided value 1`] = `
"
  max-width: 50%;
"
`;

exports[`Grid/mixins direction returns \`normal\` as the default \`flex-direction\` attribute 1`] = `
"
    flex-direction: initial;
  "
`;

exports[`Grid/mixins direction returns \`normal\` as the default \`flex-direction\` attribute 2`] = `
"
    flex-direction: initial;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 1`] = `
"
    flex-direction: column;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 2`] = `
"
    flex-direction: column-reverse;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 3`] = `
"
    flex-direction: row;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 4`] = `
"
    flex-direction: row-reverse;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 5`] = `
"
    flex-direction: row-reverse;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 6`] = `
"
    flex-direction: unset;
  "
`;

exports[`Grid/mixins direction returns the css attribute \`flex-direction\` with the provided value 7`] = `
"
    flex-direction: initial;
  "
`;

exports[`Grid/mixins gutter returns the css attributes \`padding-left\` and \`padding-right\` with the provided value 1`] = `
"
    padding-left: 1rem;
    padding-right: 1rem;
  "
`;

exports[`Grid/mixins justify returns \`normal\` as the default \`justify-content\` attribute 1`] = `
"
    justify-content: normal;
  "
`;

exports[`Grid/mixins justify returns \`normal\` as the default \`justify-content\` attribute 2`] = `
"
    justify-content: normal;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 1`] = `
"
    justify-content: flex-end;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 2`] = `
"
    justify-content: flex-start;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 3`] = `
"
    justify-content: center;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 4`] = `
"
    justify-content: space-around;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 5`] = `
"
    justify-content: space-between;
  "
`;

exports[`Grid/mixins justify returns the css attribute \`justify-content\` with the provided value 6`] = `
"
    justify-content: space-evenly;
  "
`;

exports[`Grid/mixins offset returns the css attribute \`margin-left\` with the provided value 1`] = `
"
  margin-left: 50%;
"
`;

exports[`Grid/mixins order returns the css attribute \`order\` with the provided value 1`] = `
"
    order: 1;
  "
`;

exports[`Grid/mixins verticalAlignColumns returns the css attribute \`align-items\` with the provided value 1`] = `"align-items: center"`;

exports[`Grid/mixins verticalAlignSelf returns the css attribute \`align-self\` with the provided value 1`] = `"align-self: center;"`;
